From: SeongJae Park <sj@kernel.org>
Date: Tue, 3 Feb 2026 21:37:54 -0800
Subject: [PATCH] Revert "maple_tree-add-cp_is_new_root-helper-fix"

This reverts commit 8b56f4dfdf0a4aa460fde0212eb270c943288bfa.

Without reverting, below warning is triggered on boot.

[    0.380236] ------------[ cut here ]------------
[    0.381296] WARNING: lib/maple_tree.c:2617 at mas_wr_split+0x116d/0x1270, CPU#0: swapper/0/0
[    0.383136] Modules linked in:
[    0.383797] CPU: 0 UID: 0 PID: 0 Comm: swapper/0 Not tainted 6.19.0-rc6-mm-new-damon+ #253 PREEMPT(voluntary)
[    0.385960] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 1.16.3-debian-1.16.3-2 04/01/2014
[    0.387998] RIP: 0010:mas_wr_split+0x116d/0x1270
[    0.388799] Code: ff ff ff 40 88 7b 4f 88 53 4d 4c 89 43 08 48 89 4b 10 48 89 6b 20 4c 89 63 28 e9 27 fe ff ff 4c 39 e2 0f 45 f8 e9 e7 f2 ff ff <0f> 0b e9 78 f8 ff ff 48 c7 44 24 10 00 00 00 00 b9 01 00 00 09
[    0.392744] RSP: 0000:ffffffffb2803ba0 EFLAGS: 00010006
[    0.393974] RAX: ffff8c3a0022ec00 RBX: ffffffffb2803e78 RCX: 0000000000000000
[    0.395525] RDX: ffff8c3a0022ef00 RSI: 0000000000000001 RDI: ffff8c3a0022ef50
[    0.397087] RBP: 0000000000000003 R08: 0000000000000001 R09: 0000000000000003
[    0.398627] R10: 0000000000000002 R11: ffff8c3a0022ef00 R12: ffffffffb2803c90
[    0.400179] R13: ffffffffb2803e78 R14: 0000000000000001 R15: ffffffffb2803c78
[    0.401735] FS:  0000000000000000(0000) GS:ffff8c3b38ae7000(0000) knlGS:0000000000000000
[    0.403491] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    0.404752] CR2: ffff8c3af2fff000 CR3: 0000000111224000 CR4: 00000000000000b0
[    0.405967] Call Trace:
[    0.406383]  <TASK>
[    0.406742]  mas_store_gfp+0x1f9/0x480
[    0.407497]  early_irq_init+0x135/0x170
[    0.408497]  start_kernel+0x63b/0x800
[    0.409352]  x86_64_start_reservations+0x24/0x30
[    0.410439]  x86_64_start_kernel+0xd7/0xe0
[    0.411350]  common_startup_64+0x13e/0x141
[    0.412237]  </TASK>
[    0.412710] ---[ end trace 0000000000000000 ]---

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 lib/maple_tree.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/maple_tree.c b/lib/maple_tree.c
index 56bcb307d38e2..e821040d28a45 100644
--- a/lib/maple_tree.c
+++ b/lib/maple_tree.c
@@ -2611,7 +2611,7 @@ static inline bool cp_is_new_root(struct maple_copy *cp, struct ma_state *mas)
 		 * read-side operations that can view it until it is insert into
 		 * the tree after an rcu_assign_pointer() call.
 		 */
-		ma_init_slot(cp->slot[0], cp->dst[0].node, mt);
+		RCU_INIT_POINTER(cp->slot[0], mt_mk_node(cp->dst[0].node, mt));
 		cp->height++;
 	}
 	WARN_ON_ONCE(cp->dst[0].node != mte_to_node(
-- 
2.47.3

